### PR TITLE
Use ptr.Equal to compare the value

### PR DESCRIPTION
**Pull request by Rei1010**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This uses the generic ptr.Equal in k8s.io/utils to replace functions and code constructs which only serve to return pointers to Equal values.

See https://github.com/kubernetes/utils/pull/283 for details.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by Rei1010** [comment]
@skitt ptal

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -915,11 +916,11 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 		}
 	}
 
-	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !pointer.Int32Equal(ready, job.Status.Ready)
+	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !ptr.Equal[int32](ready, job.Status.Ready)
```
Comment (skitt) [inline pkg/controller/job/job_controller.go:919]: The type can be inferred here:

```suggestion
	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !ptr.Equal(ready, job.Status.Ready)
```

File: pkg/controller/job/job_controller.go [hunk 2]
```diff
@@ -915,11 +916,11 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
...
 	job.Status.Active = active
 	job.Status.Ready = ready
 	job.Status.Terminating = jobCtx.terminating
-	needsStatusUpdate = needsStatusUpdate || !pointer.Int32Equal(job.Status.Terminating, jobCtx.terminating)
+	needsStatusUpdate = needsStatusUpdate || !ptr.Equal[int32](job.Status.Terminating, jobCtx.terminating)
```
Comment (skitt) [inline pkg/controller/job/job_controller.go:923]: Same here:

```suggestion
	needsStatusUpdate = needsStatusUpdate || !ptr.Equal(job.Status.Terminating, jobCtx.terminating)
```

**Review by skitt — COMMENTED** [review]
Thanks! You can remove a couple of type specifiers, see the comments.

While you’re at it, it seems a shame not to convert the other uses of k8s.io/utils/pointer in these three files — would you mind taking care of that too?

**Comment by skitt** [comment]
/lgtm

Thanks, FWIW (I’m not a project member).

File: pkg/apis/batch/validation/validation.go [hunk 3]
```diff
@@ -735,7 +734,7 @@ func validateCronJobSpec(spec, oldSpec *batch.CronJobSpec, fldPath *field.Path,
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.StartingDeadlineSeconds), fldPath.Child("startingDeadlineSeconds"))...)
 	}
 
-	if oldSpec == nil || !pointer.StringEqual(oldSpec.TimeZone, spec.TimeZone) {
+	if oldSpec == nil || !ptr.Equal(oldSpec.TimeZone, spec.TimeZone) {
```
Comment (thockin) [inline pkg/apis/batch/validation/validation.go:737]: You don't really need to check for nil - `ptr.Equal` does that.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**